packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2207,7 +2300,12 @@ export class KeyringController extends BaseController<
     }
 
     this.#keyrings.push(keyring);
-
+    if (
+      newKeyringMetadata &&
+      this.#keyringsMetadata.length < this.#keyrings.length
+    ) {
+      this.#keyringsMetadata = [...this.#keyringsMetadata, newKeyringMetadata];
+    }
```

Comment:
This... seeems a bit tricky to me? Why aren't we just pushing metadata with just a `this.#keyringsMetadata.push` here?

If there's a reason, we should add a comment to explain this IMO.

Reply:
For some reason this array is not always extensible 🤔 and this is a way to get around it. I was not able to find why this is happening, so I'll add a comment there

Reply:
This may be happening because we are doing something like this:
```typescript
this.#keyringsMetadata = this.state.keyringsMetadata
```
The controller state is a frozen object, and when assigning it like this to `this.#keyringsMetadata` we are only getting a pointer to the original array variable that is in the frozen state - so we get an error because in reality we are calling `.push` on the KeyringController state instead of a separate object

Reply:
That would explain a lot, however the only place where we're doing something like this is [here](https://github.com/MetaMask/core/pull/5112/files#diff-cacd62f731d7f6947fe841f6ba3441c7a16f764b699801b0c212e0e6c15879bfR665). But I already tried to change/remove this assignment and it doesn't help with the errors in the unit tests when we keep the `.push()`. So I'm still confused what is going on 😅